src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/HttpHandlerDiagnosticListener.cs

```diff
@@ -0,0 +1,711 @@
+    /// that in .NET 4.6. HttpHandlerDiagnosticListener has no public constructor. To use this, 
+    /// the application just needs to call <see cref="DiagnosticListener.AllListeners" /> and
+    /// <see cref="DiagnosticListener.AllListenerObservable.Subscribe(IObserver{DiagnosticListener})"/>,
+    /// then in the <see cref="IObserver{DiagnosticListener}.OnNext(DiagnosticListener)"/> method,
+    /// when it seems the System.Net.Http.Desktop source, subscribe to it. This will trigger the
+    /// initialization of this DiagnosticListener.
+    /// </summary>
+    internal sealed class HttpHandlerDiagnosticListener : DiagnosticListener
+    {
+        /// <summary>
+        /// Overriding base class implementation just to give us a chance to initialize.  
+        /// </summary>
+        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Predicate<string> isEnabled)
+        {
+            Initialize();
+            return base.Subscribe(observer, isEnabled);
+        }
+
+        /// <summary>
+        /// Overriding base class implementation just to give us a chance to initialize.  
+        /// </summary>
+        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Func<string, object, object, bool> isEnabled)
+        {
+            Initialize();
+            return base.Subscribe(observer, isEnabled);
+        }
+
+        /// <summary>
+        /// Initializes all the reflection objects it will ever need. Reflection is costly, but it's better to take
+        /// this one time performance hit than to get it multiple times later, or do it lazily and have to worry about
+        /// threading issues. If Initialize has been called before, it will not doing anything.
+        /// </summary>
+        private void Initialize()
+        {
+            lock (this)
+            {
+                if (!this.initialized)
+                {
+                    // This flag makes sure we only do this once. Even if we failed to initialize in an
+                    // earlier time, we should not retry because this initialization is not cheap and
+                    // the likelihood it will succeed the second time is very small.
+                    this.initialized = true;
+
+                    try
+                    {
+                        PrepareReflectionObjects();
+                        PerformInjection();
+                    }
+                    catch (Exception ex)
+                    {
+                        // If anything went wrong, just no-op. Write an event so at least we can find out.
+                        this.Write(InitializationFailed, new { Exception = ex });
+                    }
+                }
+            }
+        }
+
+#region private helper classes
+
+        [Serializable]
+        private class SimpleHashtableWrapper : Hashtable, IEnumerable
+        {
+            protected Hashtable _table;
+            public override int Count
```

Comment:
Are you allowed to use expression-bodied members in corefx? If so, these wrapper methods are all one-liners.
i.e.
```C#
public override int Count => this._table.Count;
public override bool IsReadOnly => this._table.IsReadOnly;
etc.
```